### PR TITLE
OCM-8418 | test: apply the automation labels on testing

### PR DESCRIPTION
**Pull request by yasun1**
Based on the design https://docs.google.com/document/d/1R7ZC9vJ6FsXOBA3zlQVA5jrF9hlBuwWViZMo5AOxPWk/edit, apply the automation labels on testing.

### CODE REVIEW

File: tests/ci/labels/features.go [hunk 1]
```diff
@@ -4,16 +4,57 @@ import (
...
-var FeatureKubeletConfig = Label("feature-kubeletconfig")
-var FeatureMachinepool = Label("feature-machinepool")
-var FeatureNetworkVerifier = Label("feature-networkverifier")
-var FeatureOidcConfig = Label("feature-oidcconfig")
-var FeatureOidcProvider = Label("feature-oidcprovider")
-var FeatureRegion = Label("feature-region")
-var FeatureUser = Label("feature-user")
-var FeatureToken = Label("feature-token")
-var FeatureVersion = Label("feature-version")
+// feature labels
```
Comment (radtriste) [inline tests/ci/labels/features.go:7]: Can you add a comment how this list is constructed ?
That would help to maintain it :)

File: tests/e2e/e2e_setup_test.go [hunk 2]
```diff
@@ -11,9 +11,9 @@ import (
 	"github.com/openshift/rosa/tests/utils/profilehandler"
 )
 
-var _ = Describe("ROSA CLI Test", func() {
-	It("PrepareClusterByProfile",
-		labels.Day1Prepare,
+var _ = Describe("Cluster pareparation", labels.Feature.Cluster, func() {
```
Comment (radtriste) [inline tests/e2e/e2e_setup_test.go:14]:
```suggestion
var _ = Describe("Cluster preparation", labels.Feature.Cluster, func() {
```

File: tests/e2e/e2e_setup_test.go [hunk 3]
```diff
@@ -22,10 +22,12 @@ var _ = Describe("ROSA CLI Test", func() {
 			log.Logger.Infof("Cluster prepared successfully with id %s", cluster.ID)
 		})
 
-	It("WaitClusterReady", func() {
-		clusterDetail, err := profilehandler.ParserClusterDetail()
-		Expect(err).ToNot(HaveOccurred())
-		client := rosacli.NewClient()
-		profilehandler.WaitForClusterReady(client, clusterDetail.ClusterID, config.Test.GlobalENV.ClusterWaitingTime)
-	})
+	It("to wait cluster ready",
```
Comment (radtriste) [inline tests/e2e/e2e_setup_test.go:25]:
```suggestion
	It("to wait for cluster ready",
```

File: tests/e2e/e2e_tear_down_test.go [hunk 4]
```diff
@@ -9,10 +9,9 @@ import (
 	ph "github.com/openshift/rosa/tests/utils/profilehandler"
 )
 
-var _ = Describe("ROSA CLI Test", func() {
-	It("DestroyClusterByProfile",
-		labels.Critical,
-		labels.Destroy,
+var _ = Describe("Destroy cluster", labels.Feature.Cluster, func() {
```
Comment (radtriste) [inline tests/e2e/e2e_tear_down_test.go:12]:
```suggestion
var _ = Describe("Cluster destroy", labels.Feature.Cluster, func() {
```

File: tests/e2e/hcp_cluster_test.go [hunk 5]
```diff
@@ -32,6 +32,13 @@ var _ = Describe("HCP cluster testing",
 			var err error
 			clusterConfig, err = config.ParseClusterProfile()
 			Expect(err).ToNot(HaveOccurred())
+
+			By("Skip testing if the cluster is not a HCP cluster")
+			hostedCluster, err := clusterService.IsHostedCPCluster(clusterID)
+			Expect(err).ToNot(HaveOccurred())
+			if !hostedCluster {
+				SkipTest()
```
Comment (radtriste) [inline tests/e2e/hcp_cluster_test.go:40]: I would rather a more explicit message than just `skip on profile`. The message is the results would not be explicit enough and hard to review if needed.
Or add `SkipNotHosted`, `SkipNotClassic`  methods

File: tests/e2e/e2e_skip.go [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package e2e
```
Comment (radtriste) [inline tests/e2e/e2e_skip.go:1]: Shouldn't we move this to `utils` package ?

Comment (radtriste) [reply tests/e2e/e2e_skip.go:1]: or at least rename file `utils.go` ?

File: tests/e2e/hcp_machine_pool_test.go [hunk 7]
```diff
@@ -8,58 +8,68 @@ import (
...
+	It("to hosted cluster with additional security group IDs will work [id:72195]",
+		labels.Critical, labels.Runtime.Day2,
+		func() {
+			By("Prepare security groups")
+			// This part is finding security groups according to old structure of day1.
+			// Need to be updated after ocm-common migration
+			preparedSGFile := path.Join(os.Getenv("SHARED_DIR"), "security_groups_ids")
+			if _, err := os.Stat(preparedSGFile); err != nil {
+				log.Logger.Warnf("Didn't find file %s", preparedSGFile)
+				SkipTestOnFeature("machinepool security group")
```
Comment (radtriste) [inline tests/e2e/hcp_machine_pool_test.go:40]:
```suggestion
				SkipTestOnFeature("security groups")
```

File: tests/e2e/test_rosacli_cluster.go [hunk 8]
```diff
@@ -21,8 +21,7 @@ import (
 )
 
 var _ = Describe("Edit cluster",
-	labels.Day2,
```
Comment (radtriste) [inline tests/e2e/test_rosacli_cluster.go:24]: Why not keeping the `lables.Runtime.Day2` here ? All edit are day2 operations

File: tests/e2e/test_rosacli_ocm_role.go [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-online/ocm-common/pkg/aws/aws_client"
+
+	"github.com/openshift/rosa/tests/ci/labels"
+	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+)
+
+var _ = Describe("Edit user role", labels.Feature.UserRole,
```
Comment (radtriste) [inline tests/e2e/test_rosacli_ocm_role.go:16]: This is more OCM role than user role or ?

File: tests/e2e/test_rosacli_operator_roles.go [hunk 10]
```diff
@@ -0,0 +1,358 @@
...
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/rosa/tests/ci/labels"
+	"github.com/openshift/rosa/tests/utils/common"
+	"github.com/openshift/rosa/tests/utils/config"
+	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+)
+
+var _ = Describe("Edit operator roles", labels.Feature.AccountRoles, func() {
```
Comment (radtriste) [inline tests/e2e/test_rosacli_operator_roles.go:16]:
```suggestion
var _ = Describe("Edit operator roles", labels.Feature.OperatorRoles, func() {
```

File: tests/e2e/test_rosacli_user.go [hunk 11]
```diff
@@ -10,8 +10,7 @@ import (
 )
 
 var _ = Describe("Edit User",
-	labels.Day2,
-	labels.FeatureUser,
+	labels.Feature.IDP,
```
Comment (radtriste) [inline tests/e2e/test_rosacli_user.go:13]: Shouldn't it be UserRole feature instead ?

**Review by radtriste — COMMENTED** [review]
Nice work ! See my comments below. Should we freeze the ROSA testing merge during this PR is opened ? This would avoid to have conflicts to handle for you ...
cc @xueli181114

File: tests/ci/labels/features.go (hunk 1 above)
Comment (yasun1) [reply tests/ci/labels/features.go:7]: Sure.

File: tests/e2e/test_rosacli_ocm_role.go (hunk 9 above)
Comment (yasun1) [reply tests/e2e/test_rosacli_ocm_role.go:16]: Will update.

File: tests/e2e/test_rosacli_cluster.go (hunk 8 above)
Comment (yasun1) [reply tests/e2e/test_rosacli_cluster.go:24]: There may be potential for different runtime on cases. I'd like to set the `runtime` labels on the case level so we keep the uniform rule.

File: tests/e2e/e2e_skip.go (hunk 6 above)
Comment (yasun1) [reply tests/e2e/e2e_skip.go:1]: I am thinking where could be a better place to put this, under `utils/exec` so we use it like `exec.SkipTest()`?

File: tests/e2e/hcp_cluster_test.go (hunk 5 above)
Comment (yasun1) [reply tests/e2e/hcp_cluster_test.go:40]: Yes, I have thought of users `SkipHCPTest` and `SkipClassicTest` in the past. But it is strange that we have judged `if !hostedCluster`.  
```
			if !hostedCluster {
				SkipClassicTest()
```


Seems like this is better.
```
			if !hostedCluster {
				SkipNotHosted()

Comment (radtriste) [reply tests/e2e/hcp_cluster_test.go:40]: yeah second option looks better to read

File: tests/e2e/test_rosacli_cluster.go (hunk 8 above)
Comment (radtriste) [reply tests/e2e/test_rosacli_cluster.go:24]: Ok. Please add a line to the README to set the Runtime label on the case

File: tests/e2e/test_rosacli_user.go (hunk 11 above)
Comment (radtriste) [reply tests/e2e/test_rosacli_user.go:13]: oki

File: tests/e2e/utils.go [hunk 12]
```diff
@@ -0,0 +1,21 @@
+package e2e
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+func SkipNotHosted() {
+	message := fmt.Sprintln("Skip testing on the test profile of the Classic cluster")
```
Comment (radtriste) [inline tests/e2e/utils.go:10]:
```suggestion
	message := fmt.Sprintln("The test profile is not hosted")
```

File: tests/e2e/utils.go [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+	Skip(message)
+}
+func SkipNotClassic() {
+	message := fmt.Sprintln("Skip testing on the test profile of the HCP cluster")
```
Comment (radtriste) [inline tests/e2e/utils.go:14]:
```suggestion
	message := fmt.Sprintln("The test profile is not classic")
```

File: tests/e2e/utils.go [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+	Skip(message)
+}
+
+func SkipTestOnFeature(feature string) {
+	message := fmt.Sprintf("Skip! The test profile is not configured with the feature: %s", feature)
```
Comment (radtriste) [inline tests/e2e/utils.go:19]:
```suggestion
	message := fmt.Sprintf("The test profile is not configured with the feature: %s", feature)
```

File: tests/ci/labels/features.go (hunk 1 above)
Comment (radtriste) [reply tests/ci/labels/features.go:7]: Please add a description here !

**Review by radtriste — APPROVED** [review]
/lgtm

**Pull request merged**